kafka-client/src/main/java/dev/responsive/kafka/clients/ResponsiveKafkaClientSupplier.java

```diff
@@ -54,7 +56,7 @@
  * close callbacks on the returned clients). As these calls are not on the performance path we
  * rely on coarse-grained locks around sections reading/writing shared state.
  */
-public final class ResponsiveKafkaClientSupplier implements KafkaClientSupplier, Configurable {
+public final class ResponsiveKafkaClientSupplier implements KafkaClientSupplier {
```

Comment:
isn't it necessary to make this `Configurable` if they want to pass it in using the `default.client.supplier` config?

Reply:
I don't think there is a `default.client.supplier` config? At least not in Streams, so the way it is right now it's never getting configured at all unless explicitly invoked.

Anyways part of this PR and the new API improvements are that we pass in our client supplier ourselves rather than making the user pass it in, however that would be. Whether or not there's a config for this, you can definitely pass it in via the `KafkaStreams` constructor, so that's what we're now doing in `ResponsiveKafkaStreams`